test/mavsdk_tests/autopilot_tester.h

```diff
@@ -128,24 +128,31 @@ class AutopilotTester
 			uint32_t start_time = _info->get_flight_information().second.time_boot_ms;
 
 			while (!fun()) {
-				std::this_thread::sleep_for(duration_ms / 1000);
+				std::this_thread::sleep_for(duration_ms / 100);
 
 				// This might potentially loop forever and the test needs to be killed by a watchdog outside.
 				// The reason not to include an absolute timeout here is that it can happen if the host is
 				// busy and PX4 doesn't run fast enough.
-				if (_info->get_flight_information().second.time_boot_ms - start_time > duration_ms.count()) {
+				const int64_t elapsed_time_ms = _info->get_flight_information().second.time_boot_ms - start_time;
+
+				if (elapsed_time_ms > duration_ms.count()) {
+					std::cout << "Timeout, connected to vehicle but waiting for test for " << elapsed_time_ms / 1000.0 << " seconds" <<
+						  std::endl;
 					return false;
 				}
 			}
 
 		} else {
 			// Nothing is connected yet. Use the host time.
-			unsigned iteration = 0;
+			const auto start_time = std::chrono::steady_clock::now();
 
 			while (!fun()) {
-				std::this_thread::sleep_for(duration_ms / 1000);
+				std::this_thread::sleep_for(duration_ms / 100);
+				const int64_t elapsed_time_us = std::chrono::duration<double, std::micro>(std::chrono::steady_clock::now() -
+								start_time).count();
```

Comment:
I'm confused about the types here. 

What about?

```suggestion
				std::this_thread::sleep_for(std::chrono::duration_cast<std::chrono::microseconds>(duration_ms)/ 100);
				const auto elapsed_time_ms = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - start_time);
```